ptclient/ldap.c

```diff
@@ -1173,9 +1176,9 @@ static int ptsmodule_make_authstate_attribute(
                 }
                 strlcat((*newstate)->groups[i].id, rdn[0], sizeof((*newstate)->groups[i].id));
                 (*newstate)->groups[i].hash = strhash((*newstate)->groups[i].id);
+                ldap_value_free(rdn);
```

Comment:
`rdn` is declared in the function scope, but it looks as if there is no need for that. It could instead be declared inside the block of this for loop.

Reply:
I agree -- I initially didn't do this cause when I started looking at the variable scoping generally, it's all a huge mess, and I didn't want to complicate things by refactoring it all.  But there's no reason I can't just fix this one variable while I'm looking directly at it, so I'll do so!